### PR TITLE
treewide: use lib.optionalAttrs

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/build-managers/bazel/bazel_5/default.nix [hunk 1]
```diff
@@ -282,10 +282,10 @@ stdenv.mkDerivation rec {
         sha256 = "1mm4awx6sa0myiz9j4hwp71rpr7yh8vihf3zm15n2ii6xb82r31k";
       };
 
-    in (if !stdenv.hostPlatform.isDarwin then {
+    in (lib.optionalAttrs (!stdenv.hostPlatform.isDarwin) {
       # `extracted` doesn’t work on darwin
       shebang = callPackage ../shebang-test.nix { inherit runLocal extracted bazelTest distDir; bazel = bazel_self;};
-    } else {}) // {
+    }) // {
```
Comment (Ericson2314) [inline pkgs/development/tools/build-managers/bazel/bazel_5/default.nix:288]: Parens like these will no longer be needed with this change.

File: pkgs/build-support/release/source-tarball.nix [hunk 2]
```diff
@@ -117,7 +117,7 @@ stdenv.mkDerivation (
       version = version + versionSuffix;
     };
 
-    meta = (if args ? meta then args.meta else {}) // {
+    meta = (lib.optionalAttrs (args ? meta) args.meta) // {
```
Comment (kilianar) [inline pkgs/build-support/release/source-tarball.nix:120]: `lib` isn't passed as an argument, causing the build to fail:
```
error: undefined variable 'lib'

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/pkgs/build-support/release/source-tarball.nix:120:13:

          119|
          120|     meta = (lib.optionalAttrs (args ? meta) args.meta) // {
             |             ^
          121|       description = "Source distribution";

       … while calling 'sourceTarball'
```

Comment (Stunkymonkey) [reply pkgs/build-support/release/source-tarball.nix:120]: I removed my changes to this file. No idea how to import lib here.

Comment (winterqt) [reply pkgs/build-support/release/source-tarball.nix:120]: You add it here: https://github.com/NixOS/nixpkgs/blob/4f1b2d513327b121cbf82b4086b3b854a48347a3/pkgs/build-support/release/default.nix#L8

**Review by winterqt — CHANGES_REQUESTED** [review]
The commit message has a typo, can you fix it?

File: pkgs/build-support/release/source-tarball.nix (hunk 2 above)
Comment (Stunkymonkey) [reply pkgs/build-support/release/source-tarball.nix:120]: I tried that before. Have a look at the error.

Comment (winterqt) [reply pkgs/build-support/release/source-tarball.nix:120]: Try replacing the reference to `args` in this line with `(builtins.removeAttrs args [ "lib" ])`: https://github.com/NixOS/nixpkgs/blob/63b5955814db30d2e2ff7157aaa5665b502ed2f4/pkgs/build-support/release/source-tarball.nix#L76

**Comment by Stunkymonkey** [comment]
ping @winterqt ?

**Comment by Stunkymonkey** [comment]
ping @SuperSandro2000

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by Stunkymonkey** [comment]
fixed the conflicts and rebased.

**Pull request merged**